workflow/validate/validate.go

```diff
@@ -82,6 +82,10 @@ const (
 	anyWorkflowOutputParameterMagicValue = "workflow.outputs.parameters.*"
 	anyWorkflowOutputArtifactMagicValue  = "workflow.outputs.artifacts.*"
 	maxCharsInObjectName                 = 63
+	// CronWorkflows have fewer max chars allowed in their name because when workflows are created from them, they
+	// are appended with the unix timestamp (`-1615836720`). This lower character allowance allows for that timestamp
+	// to still fit within the 63 character maximum.
+	maxCharsInCronWorkflowName           = 52
```

Comment:
This will technically need to be reduced to `51` on `Saturday November 20 2286 17:46:40 GMT` to accommodate the extra character required by the Unix timestamp, but I think we should be okay.